apis/pkg/v1beta1/image_config_types.go

```diff
@@ -78,12 +94,199 @@ type RegistryConfig struct {
 	Authentication *RegistryAuthentication `json:"authentication,omitempty"`
 }
 
-// ImageConfigSpec contains the configuration for matching images.
-type ImageConfigSpec struct {
-	// MatchImages is a list of image matching rules that should be satisfied.
-	// +kubebuilder:validation:XValidation:rule="size(self) > 0",message="matchImages should have at least one element."
-	MatchImages []ImageMatch `json:"matchImages"`
-	// Registry is the configuration for the registry.
+// ImageVerification contains the configuration for verifying the image.
+type ImageVerification struct {
+	// Provider is the provider that should be used to verify the image.
+	// +kubebuilder:validation:Enum=Cosign
+	Provider string `json:"provider"`
+	// Cosign is the configuration for verifying the image using cosign.
 	// +optional
-	Registry *RegistryConfig `json:"registry,omitempty"`
+	Cosign *CosignVerificationConfig `json:"cosign,omitempty"`
+}
+
+// CosignVerificationConfig contains the configuration for verifying the image
+// using cosign.
+type CosignVerificationConfig struct {
+	// Authorities defines the rules for discovering and validating signatures.
+	Authorities []CosignAuthority `json:"authorities"`
+}
+
+// Copied from https://github.com/sigstore/policy-controller/blob/d73e188a4669780af82d3d168f40a6fff438345a/pkg/apis/policy/v1alpha1/clusterimagepolicy_types.go#L118
```

Comment:
https://github.com/sigstore/policy-controller has apache 2.0 [license](https://github.com/sigstore/policy-controller), wondering whether I still need to include some copyright notice somewhere still 🤔

Reply:
Copied the license header with a note, I believe this should be enough: https://github.com/crossplane/crossplane/pull/6022/files#diff-3081c28ab15eb783d9e4b075756801e3914fc9cb353558404e203afae15a7912R1-R19